web/src/components/__tests__/Feedback.spec.tsx

```diff
@@ -0,0 +1,133 @@
+    isPositiveFeedback: boolean,
+    comment: string,
+    isSearchFeedback = false,
+    sendingStatus = SendingState.IDLE
+  ) => {
+    return {
+      comment,
+      isPositiveFeedback,
+      isSearchFeedback,
+      contactMail: 'test@example.com',
+      sendingStatus,
+      onCommentChanged,
+      onContactMailChanged,
+      onSubmit,
+      t,
+      closeFeedbackModal
+    }
+  }
+
+  it('button should be disabled for negative Feedback and no input', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={buildConfig().lightTheme}>
+        <Feedback {...buildProps(false, '')} />
+      </ThemeProvider>
+    )
+    expect(getByText('feedback:send')).toBeDisabled()
+  })
+
+  it('button should be enabled for positive Feedback and no input', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={buildConfig().lightTheme}>
+        <Feedback {...buildProps(true, '')} />
+      </ThemeProvider>
+    )
+    expect(getByText('feedback:send')).not.toBeDisabled()
+  })
+
+  it('button should be enabled for negative Feedback and input', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={buildConfig().lightTheme}>
+        <Feedback {...buildProps(false, 'comment')} />
+      </ThemeProvider>
+    )
+    expect(getByText('feedback:send')).not.toBeDisabled()
+  })
+
+  it('should display correct description for search', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={buildConfig().lightTheme}>
+        <Feedback {...buildProps(false, 'comment', true)} />
+      </ThemeProvider>
+    )
+    expect(getByText('feedback:wantedInformation')).toBeTruthy()
+  })
+
+  it('should display error', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={buildConfig().lightTheme}>
+        <Feedback {...buildProps(false, 'comment', false, SendingState.ERROR)} />
+      </ThemeProvider>
+    )
+    expect(getByText('feedback:failedSendingFeedback')).toBeTruthy()
+  })
+
+  it('onSubmit should be called on button press', async () => {
+    const { getByText } = render(
+      <ThemeProvider theme={buildConfig().lightTheme}>
+        <Feedback {...buildProps(false, 'comment')} />
+      </ThemeProvider>
+    )
+    const button = getByText('feedback:send')
+    fireEvent.click(button)
+    expect(onSubmit).toBeCalled()
+  })
+
+  it('should call callback on contact mail changed', () => {
+    const { getByDisplayValue, queryByDisplayValue } = render(
+      <ThemeProvider theme={buildConfig().lightTheme}>
+        <Feedback {...buildProps(false, 'my comment')} />
+      </ThemeProvider>
+    )
+    expect(getByDisplayValue('test@example.com')).toBeTruthy()
+    expect(queryByDisplayValue('new@example.com')).toBeFalsy()
+    expect(onContactMailChanged).not.toHaveBeenCalled()
+    fireEvent.change(getByDisplayValue('test@example.com'), {
+      target: {
+        value: 'new@example.com'
+      }
+    })
+    expect(onContactMailChanged).toHaveBeenCalledTimes(1)
```

Comment:
I thinkg `onContactMailChanged` is created only once for the whole testsuite. that means if you also call onContactMailChanged is some other test within this suite, then this test will fail.

I'm not 100% sure whether this is true, because jest is initializing tests is a weird way. But I think this test should be more isolated from the other tests by calling `jest.fn()` from within the `should call callback on contact mail changed` test.